File: LC_JS-SOLUTIONS/49. Group Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {

    strDic = {}
    res = []
    for(let index in strs){
        //console.log(strs[index].split('').sort().join(''))
        if(!(strs[index].split('').sort().join('') in strDic)){
            strDic[strs[index].split('').sort().join('')] = [strs[index]]
        }
        else{
            strDic[strs[index].split('').sort().join('')].push(strs[index])
         }
    }

    for (index in strDic){
        res.push(strDic[index])
    }

    return(res)
    
};

File: LC-Python-Solution/49. Group Anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        res = []
        strDic = {}

        for subStr in strs:
            if(''.join(sorted(subStr)) not in strDic):
                strDic[''.join(sorted(subStr))] = [subStr]
            
            else:
                strDic[''.join(sorted(subStr))].append(subStr)
        
        for index,key in enumerate(strDic):
            res.append(strDic[key])
        
        return res





        